vm/src/lib.rs

```diff
@@ -111,6 +118,46 @@ impl VM {
         Ok(vm)
     }
 
+    /// Sets the activation height for a specific host query.
+    ///
+    /// This method associates a previously registered host query with a block
+    /// height at which it becomes active. Before this activation height,
+    /// the host query will be excluded from session execution.
+    ///
+    /// **Note:** The specified host query must already be registered in the
+    /// global host queries registry before calling this method.
+    ///
+    /// # Arguments
+    /// * `host_query` - The name of the host query to activate.
+    /// * `activation` - The block height at which the host query becomes
+    ///   active.
+    ///
+    /// # Panics
+    /// This method will panic if the provided `host_query` is not already
+    /// registered in the global host queries registry.
+    ///
+    /// # Examples
+    /// ```rust
+    /// use dusk_vm::VM;
+    /// use dusk_core::abi::Query;
+    ///
+    /// let mut vm = VM::ephemeral().unwrap();
+    /// vm.with_hq_activation(Query::KECCAK256, 100);
+    /// ```
+    pub fn with_hq_activation<S: Into<String>>(
+        &mut self,
+        host_query: S,
+        activation: u64,
+    ) {
+        let host_query = host_query.into();
+        if self.inner.host_queries().get(&host_query).is_none() {
+            panic!(
+                "Host query '{host_query}' must be registered before setting activation"            
+            );
+        }
+        self.hq_activation.insert(host_query, activation);
```

Comment:
If the `host_query` is already present in the `HashMap` the block-height will be updated to the new value, is this something we want to prevent? Or at least inform the user about (by returning the old value if present and `None` if not) just like the `insert` method does?

Reply:
Good question
My answer is "I don't know"

This is simply populated by config, and should not be played around by the user